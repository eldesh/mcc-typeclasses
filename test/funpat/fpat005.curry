-- This example combines function patterns with constrained variables
-- that can occur more than once

g x y z | y =/= z = (const z x,const z y)
f (g x 1 y) = [x,y]

main = f (unknown,unknown)
