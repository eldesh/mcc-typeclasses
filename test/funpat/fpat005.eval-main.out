{_a /= 1} [_b,_a]
